engine/src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -180,59 +181,73 @@ trait CromwellApiService {
         }
       }
     } ~
-  path("workflows" / Segment) { version =>
+      path("workflows" / Segment) { version =>
       post {
         entity(as[Multipart.FormData]) { formData =>
-          submitRequest(formData, true)
+          submitRequest(formData, isSingleSubmission = true, version)
         }
       }
     } ~
-  path("workflows" / Segment / "batch") { version =>
+      path("workflows" / Segment / "batch") { version =>
     post {
       entity(as[Multipart.FormData]) { formData =>
-        submitRequest(formData, false)
+        submitRequest(formData, isSingleSubmission = false, version)
       }
     }
   }
 
-  private def submitRequest(formData: Multipart.FormData, isSingleSubmission: Boolean): Route = {
+  private def submitRequest(formData: Multipart.FormData, isSingleSubmission: Boolean, version: String): Route = {
+    val versionRegex = """v(\d+)""".r
+    version match {
+      case versionRegex(number) if number == "1" || number == "2" =>
+        submitRequest(formData, isSingleSubmission, number.toInt)
+      case _ => reject
+    }
+  }
+
+  private def submitRequest(formData: Multipart.FormData, isSingleSubmission: Boolean, version: Int): Route = {
     val allParts: Future[Map[String, ByteString]] = formData.parts.mapAsync[(String, ByteString)](1) {
-      case b: BodyPart => b.toStrict(duration).map(strict => b.name -> strict.entity.data)
+      bodyPart => bodyPart.toStrict(duration).map(strict => bodyPart.name -> strict.entity.data)
     }.runFold(Map.empty[String, ByteString])((map, tuple) => map + tuple)
 
     def toResponse(workflowId: WorkflowId): WorkflowSubmitResponse = {
       WorkflowSubmitResponse(workflowId.toString, WorkflowSubmitted.toString)
     }
 
-    def askSubmit(command: WorkflowStoreActor.WorkflowStoreActorSubmitCommand): Route = {
+    def askSubmit(command: WorkflowStoreActor.WorkflowStoreActorSubmitCommand, warnings: Seq[String]): Route = {
       // NOTE: Do not blindly coppy the akka-http -to- ask-actor pattern below without knowing the pros and cons.
```

Comment:
Also don't blindly copy this spelling of copy.

Reply:
Hehe, was hoping no one noticed that. I'm going to leave it as a "watermark".